src/server/journal/streamer.cc

```diff
@@ -40,4 +49,127 @@ void JournalStreamer::WriterFb(io::Sink* dest) {
   }
 }
 
+RestoreStreamer::RestoreStreamer(DbSlice* slice, SlotSet slots, journal::Journal* journal,
+                                 Context* cntx)
+    : JournalStreamer(journal, cntx), db_slice_(slice), my_slots_(std::move(slots)) {
+  DCHECK(slice != nullptr);
+}
+
+void RestoreStreamer::Start(io::Sink* dest) {
+  auto db_cb = absl::bind_front(&RestoreStreamer::OnDbChange, this);
+  snapshot_version_ = db_slice_->RegisterOnChange(std::move(db_cb));
+
+  JournalStreamer::Start(dest);
+
+  DCHECK(!snapshot_fb_.IsJoinable());
+  snapshot_fb_ = fb2::Fiber("slot-snapshot", [this] {
+    PrimeTable::Cursor cursor;
+    uint64_t last_yield = 0;
+    PrimeTable* pt = &db_slice_->databases()[0]->prime;
+
+    do {
+      if (fiber_cancellation_.IsCancelled())
+        return;
+
+      cursor = pt->Traverse(cursor, absl::bind_front(&RestoreStreamer::WriteBucket, this));
+
+      if (last_yield >= 100) {
+        ThisFiber::Yield();
+        last_yield = 0;
+      }
+    } while (cursor);
+
+    WriteCommand(make_pair("DFLYMIGRATE", ArgSlice{"FULL-SYNC-CUT"}));
```

Comment:
I think we need to discuss it tomorrow. we need to add sync_id argument and this command should be sent only once for all flows